File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    answer = 0
    queue = [(p, idx) for idx, p in enumerate(priorities)]

    while queue:
        current_priority, current_idx = queue.pop(0)
        if any (current_priority < p for p, idx in queue):
            queue.append((current_priority, current_idx))
        else:
            answer += 1
            if location == current_idx:
                break
        
    return answer

File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        
        Queue<Integer> queue = new LinkedList();
        List<Integer> alphabetIdx = new ArrayList<>();
        
        for (int i = 0; i < priorities.length; i++) {
            queue.add(priorities[i]);
            alphabetIdx.add(i);
        }
        
        while (!queue.isEmpty()) {
            int currentPriority = queue.poll();
            int currentIdx = alphabetIdx.remove(0);
            
            boolean hasHigher = false;
            for (int priority : queue) {
                if (priority > currentPriority) {
                    hasHigher = true;
                    break;
                }
            }
            
            // 더 높은 우선순위를 가진 값이 있다면
            if (hasHigher) {
                queue.add(currentPriority);
                alphabetIdx.add(currentIdx);
            } else {
                answer++;
                
                if (currentIdx == location) {
                    return answer;
                }
            }
        }
        
        return answer;
    }
}